src/hotspot/share/gc/shared/genArguments.cpp

```diff
@@ -37,7 +37,11 @@ size_t MinNewSize = 0;
 size_t MinOldSize = 0;
 size_t MaxOldSize = 0;
 
-size_t OldSize = 0;
+// If InitialHeapSize or MinHeapSize is not set on cmdline, this variable,
+// together with NewSize, is used to derive them.
+// Using the same value when it was a configurable flag to avoid breakage.
+// See more in JDK-8345323
```

Comment:
I do not like referrals to the bug tracker in the code, and/or referring to some code the past ("when it was configurable").
Better explicitly state the problem with heap sizing and large pages and file a follow-up RFE (not mentioning it here).

Reply:
I.e. something like "If the default value of OldSize is too small, then ..., leading to the generations not aligned and not being able to allocate large pages" or so.